### ISSUE TITLE
[Bug Report] Unable to extract archive

### ISSUE THREAD

**Issue by dev-mcheung** [hkalexling/Mango#90]
**Describe the bug**
When Mango fails to extract a file; the download will stall with no indication of what is going on for the end users. 

**To Reproduce**
1. Go to 'Download'
2. Use 'https://mangadex.org/title/7139/one-punch-man'
3. Select group 'r/Onepunchman'
4. 'Select all' 
5. Process download 

**Expected behavior**
This downloads about two chapters before chapter 101 stalls. I checked the logs and apparently there is an error.

`Unable to extract archive /root/mango/library/One Punch-Man/Ch.101 Zombieman.cbz. Ignoring it. Archive error: Couldn't find directory end signature in the last 65KB`

We need some kind of an error status on the frontend letting the user know. At the moment it is just stuck on "Downloading" while the backend tries to extract the files in a loop every time the scan is run.

**Environment (please complete the following information):**
 - Tested in two different OS: Unraid v6.8.1 and Ubuntu 18.04.2 LTS 
 - Browser: N/A
 - Mango Version 0.7.3

**Comment by hkalexling** [comment]
The archive error is expected, because the archive is still being downloaded and therefore it's not a valid zip archive. The error should go away after the download is completed. 

You mention that the download stalls. Do you see any download error in the log?

**Comment by dev-mcheung** [comment]
There is no download error. Just the below error:

`Unable to extract archive /root/mango/library/One Punch-Man/Ch.101 Zombieman.cbz. Ignoring it. Archive error: Couldn't find directory end signature in the last 65KB`

**Comment by hkalexling** [comment]
The fact that the cbz file is present on your file system does not mean it has finished downloading. When downloading a chapter, Mango does the followings

1. Create an empty cbz file on your system
2. Download each page and insert the page to the cbz file
3. Close the cbz file

The cbz file would be invalid until step 3 is done. In your case, Mango was scanning the library during step 2, and that's why you saw the archive error. Again, this is completely normal, and the error goes away when you rescan after step 3. 

I tried downloading the same chapter while scanning the library, and the download finished normally. Since there was no download error, I guess the "stalling" was due to slow connection. Could you try and download it again?

> Regardless, the client should still let the user know that the file is corrupted and offer an option to retry the download.

I agree, and this has been requested in #49.

**Comment by jaredlt** [comment]
I wonder if it's worth creating the file initially with an unsupported file extension so that Mango doesn't consider it when doing the scan? Eg. `Vol 1.cbz.tmp` (or without the .`cbz` completely if that would still confuse things). I've seen other download clients do something similar.

**Comment by dev-mcheung** [comment]
So I just deleted the .cbz from the directory and also cleared the download task for Chapter 101. The other downloads do automatically resume either. 'Refresh queue' and 'Pause/resume download' doesn't seem to restart the tasks either. I'll probably reference this one as a new issue.

**Comment by hkalexling** [comment]
@jaredlt Good idea! I like `.part` (that's how Firefox names partially downloaded files) as it's more explicit than `.tmp`. I will try to include this in the next release. 

@dev-mcheung What are their statuses in the download manager? It might be helpful if you set the log level to `debug`.

**Comment by dev-mcheung** [comment]
The overall fix for this was to restart the docker. Download manager started to work right afterwards. Chapter 101 also downloaded properly for me as well.
